src/background.js

```diff
@@ -19,12 +21,19 @@ import { isValidUrl } from './url'
 
 const CHECK_INTERVAL_TIMER = 60000
 
-chrome.runtime.onMessage.addListener(async (message, sender, sendResponse) => {
+chrome.runtime.onMessage.addListener((message, sender, sendResponse) => {
   const { action, msg: badgeCounter } = message
   switch (action) {
     case ACTION_UPDATE_BADGE_COUNTER:
       updateBadgeCounterUI(badgeCounter)
       break
+    case ACTION_GET_CURRENT_TAB_URL: {
+      getCurrentTabUrlBackground().then(tabUrl => {
+        sendResponse(tabUrl)
+      })
+      // This is required when we want to call sendResponse callback asynchronously
+      return true
```

Comment:
why does this have to be done this way?

Reply:
According to the docs https://developer.chrome.com/docs/extensions/mv3/messaging/#simple - listener must return true if we want to call `sendResponse` asynchronously.

Reply:
@TMaszko be careful that we are not using `content_scripts`: https://developer.chrome.com/docs/extensions/mv3/content_scripts/ 
We are not using those: https://developer.chrome.com/docs/extensions/mv3/content_scripts/#static-declarative
and that's why we don't need anymore the `activeTab`.
We are just using `popup` page and `background` service worker.

During your tests is this implementation working as expected?

Reply:
@davideroffo I've tested it and it worked properly. Moreover https://developer.chrome.com/docs/extensions/reference/runtime/ here there is a info about `popups` as well
"These methods support message passing so that you can communicate with different parts of your extension (such as an extension `popup` and `background` scripts), other extensions, or native applications on the user's device. See [Message Passing](https://developer.chrome.com/docs/extensions/mv3/messaging/) for an overview of the subject. Methods in this category include [connect](https://developer.chrome.com/docs/extensions/reference/runtime/#method-connect), [connectNative](https://developer.chrome.com/docs/extensions/reference/runtime/#method-connectNative), [sendMessage](https://developer.chrome.com/docs/extensions/reference/runtime/#method-sendMessage), and [sendNativeMessage](https://developer.chrome.com/docs/extensions/reference/runtime/#method-sendNativeMessage)."